tests/sweep_framework/sweeps/line_all_gather.py

```diff
@@ -0,0 +1,106 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Optional, Tuple
+
+import torch
+
+import ttnn
+
+from tests.ttnn.utils_for_testing import check_with_pcc, start_measuring_time, stop_measuring_time
+from loguru import logger
+import tt_lib as ttl
+import ttnn
+from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import comp_equal, comp_pcc
+from tests.ttnn.unit_tests.operations.test_all_gather import is_unsupported_case
+from ttnn import ShardTensorToMesh
+
+# Override the default timeout in seconds for hang detection.
+TIMEOUT = 30
+
+# Parameters provided to the test vector generator are defined here.
+# They are defined as dict-type suites that contain the arguments to the run function as keys, and lists of possible inputs as values.
+# Each suite has a key name (in this case "suite_1" and "suite_2") which will associate the test vectors to this specific suite of inputs.
+# Developers can create their own generator functions and pass them to the parameters as inputs.
+parameters = {
+    "suite_1": {
+        "num_devices": [4, 8],
+        "num_links": [1, 2],
+        "input_shape": [
+            [8, 1, 33, 256],
+            [8, 1, 256, 32],
+            [8, 8, 256, 384],
+            [8, 5, 13, 512],
+            [8, 5, 32, 512],
+            [1, 1, 32, 16384],
+        ],
+        "dim": [0, 1, 3],
+        "layout": [ttl.tensor.Layout.ROW_MAJOR, ttl.tensor.Layout.TILE],
+        "input_dtype": [ttl.tensor.DataType.BFLOAT16],
+        "mem_config": [ttl.tensor.MemoryConfig(buffer_type=ttl.tensor.BufferType.DRAM)],
+        "enable_async": [True, False],
+    },
+}
+
+
+def skip(
+    *, t3k_device_mesh, input_shape, dim, mem_config, num_devices, num_links, input_dtype, layout, **_
+) -> Tuple[bool, Optional[str]]:
+    if t3k_device_mesh.get_num_devices() != 8:
+        return True, "Not T3000!"
+    (is_known_failure, message) = is_unsupported_case(
+        input_shape, dim, mem_config, num_devices, num_links, input_dtype, layout
+    )
+    if is_known_failure:
+        return True, f"Skipping unsupported case {message}."
+    return False, None
+
+
+# This is the run instructions for the test, defined by the developer.
+# The run function must take the above-defined parameters as inputs.
+# The runner will call this run function with each test vector, and the returned results from this function will be stored.
+def run(
```

Comment:
Required inputs to this run function must match the defined parameters above, make sure all inputs are defined in the parameters list so the test can run with the sweeps infra.

Reply:
For the device setup, please see this new PR and accompanying README change. This will allow you to define the device mesh you need for your test: https://github.com/tenstorrent/tt-metal/pull/11486

You should copy the device fixture code you require to a generator function called `device_mesh_fixture` in this file. The sweep infra will pick this up, use it, and pass it to the test in the `device` parameter which you should include instead of t3k_device_mesh here.

Reply:
Thanks for the PR, I will look into it and try adding device generator